pkg/api/svc.go

```diff
@@ -117,6 +136,20 @@ func (s *Service) Handle(conn wasabi.Connection, r wasabi.Request) error {
 	}
 }
 
+// getRequestLimits is a helper function transform request limits mentioned in server configuration
+// into wasabi request limits, so that we can use the RateLimiter middleware.
+func getRequestLimits(rateLimitCfg RateLimits) func(wasabi.Request) (string, time.Duration, uint64) {
+	return func(r wasabi.Request) (string, time.Duration, uint64) {
+		duration, err := time.ParseDuration(rateLimitCfg.General.Interval)
+		if err != nil {
+			//TODO: should we panic or fallback to defaults with warning?
+			panic(errors.New("incorrect interval format"))
+		}
+
+		return r.RoutingKey(), duration, uint64(rateLimitCfg.General.Limit)
```

Comment:
your rate limiting key should be IP + name of category like(general, group 1 or group 2)

for getting ip you can use this middleware https://github.com/ksysoev/wasabi/blob/main/middleware/http/clientip.go , we can make Cloudflare as default provider, or set an option in config file

Reply:
latter probably would be nice to add getter of ip from context into wasabi middleware